src/main/java/edu/greenblitz/robotName/subsystems/LED/LED.java

```diff
@@ -0,0 +1,140 @@
+package edu.greenblitz.robotName.subsystems.LED;
+
+import edu.greenblitz.robotName.OI;
+import edu.greenblitz.robotName.ScoringModeSelector;
+import edu.greenblitz.robotName.subsystems.intake.Intake;
+import edu.greenblitz.robotName.subsystems.arm.roller.Roller;
+import edu.greenblitz.robotName.subsystems.shooter.funnel.Funnel;
+import edu.greenblitz.robotName.utils.GBSubsystem;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+
+import static edu.greenblitz.robotName.subsystems.LED.LEDConstants.*;
+
+public class   LED extends GBSubsystem {
+	
+	private Color currentColor;
+	private static LED instance;
+	private AddressableLED addressableLED;
+	private AddressableLEDBuffer addressableLEDBuffer;
+	private Timer LEDBlinkTimer;
+	private boolean lastNotePosition;
+
+	private SendableChooser<Boolean> chooser;
+
+	private LED() {
+		this.addressableLED = new AddressableLED(LED_PORT);
+		this.addressableLEDBuffer = new AddressableLEDBuffer(LED_LENGTH);
+		this.addressableLED.setLength(LED_LENGTH);
+		this.addressableLED.start();
+		LEDBlinkTimer = new Timer();
+		LEDBlinkTimer.restart();
+		currentColor = SPEAKER_MODE_COLOR;
+		lastNotePosition = false;
+
+		chooser = new SendableChooser<>();
+		chooser.setDefaultOption("False", false);
+		chooser.addOption("True", true);
+		SmartDashboard.putData("Is npot robot", chooser);
+	}
+	
+	public static LED getInstance() {
+		init();
+		return instance;
+	}
+	
+	public static void init() {
+		if (instance == null) {
+			instance = new LED();
+		}
+	}
+	
+	public double getLEDBlinkTimer() {
+		return LEDBlinkTimer.get();
+	}
+
+	public boolean getLastNotePosition(){
+		return lastNotePosition;
+	}
+	
+	public void restartTimer() {
+		LEDBlinkTimer.restart();
+	}
+	
+	
+	public void setLEDColor(Color color, Section section) {
+		setLEDColor(color, section.startIndex(), section.endIndex());
+	}
+	
+	public void setLEDColor(Color color, int startIndex, int endIndex) {
+		for (int i = startIndex; i < endIndex; i++) {
+			setLEDColor(color, i);
+		}
+	}
+	
+	public void setLEDColor(Color color, int index) {
+		this.addressableLEDBuffer.setLED(index, color);
+	}
+	
+	public void turnOff(int index) {
+		setLEDColor(new Color(0, 0, 0), index);
+	}
+	
+	public void turnOff(int startIndex, int endIndex) {
+		for (int i = startIndex; i < endIndex; i++) {
+			turnOff(i);
+		}
+	}
+	
+	public void turnOff(Section section) {
+		turnOff(section.startIndex(), section.endIndex());
+	}
```

Comment:
this is kinda confusing, i think this is fine but i would prefer there would be seperate names for the functions instead of using the same ones

Reply:
romy - they both do the same thing tho, its fine